src/sas/sasgui/perspectives/fitting/fitpage.py

```diff
@@ -1235,10 +1235,13 @@ def _on_select_model(self, event=None):
             new_event = PageInfoEvent(page=self)
             wx.PostEvent(self.parent, new_event)
             # update list of plugins if new plugin is available
-            custom_model = CUSTOM_MODEL
             mod_cat = self.categorybox.GetStringSelection()
-            if mod_cat == custom_model:
+            if mod_cat == CUSTOM_MODEL:
+                temp_id = self.model.id
                 temp = self.parent.update_model_list()
+                for v in self.parent.model_dictionary.values():
+                    if v.id == temp_id:
+                        self.model = v()
```

Comment:
This seems like a single model is desired. A break here would exit out of the for loop as soon as it is found.